File: LeetCode/python3/9.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0 or (x != 0 and x % 10 == 0):
            return False
        a, b = x, 0
        while a > b:
            b = b * 10 + a % 10
            a //= 10
        return a == b or a == b // 10

File: LeetCode/cpp/9.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if (x < 0)
            return false;
        vector<int> nums;
        while (x) {
            nums.push_back(x % 10);
            x /= 10;
        }
        int i = 0, j = nums.size() - 1;
        while (i < j) {
            if (nums[i] != nums[j])
                return false;
            ++i, --j;
        }
        return true;
    }
};
